src/Microsoft.Sbom.Parsers.Spdx30SbomParser/Utils/SPDXToSbomFormatConverterExtensions.cs

```diff
@@ -15,36 +18,202 @@ namespace Microsoft.Sbom.Utils;
+            Path = spdxFile.Name,
+            Id = spdxFile.SpdxId
+        };
+
+        return sbomFile;
+    }
+
     /// <summary>
     /// Converts a <see cref="SPDXFile"/> object to a <see cref="SbomFile"/> object.
     /// </summary>
     /// <param name="spdxFile"></param>
     /// <returns></returns>
-    public static SbomFile ToSbomFile(this File spdxFile)
+    public static SbomFile ToSbomFile(this File spdxFile, List<Element> spdx30Elements, List<Common.Spdx30Entities.Relationship> relationships)
     {
-        var checksums = spdxFile.VerifiedUsing?.Select(c => c.ToSbomChecksum());
+        // Note that the SPDX 3.0 HAS_DECLARED_LICENSE relationship type is equivalent to LicenseInfoInFiles internally.
+        var sbomFile = new SbomFile
+        {
+            Checksum = spdxFile.VerifiedUsing.ToSbomChecksum(),
+            FileCopyrightText = spdxFile.CopyrightText == "NOASSERTION" ? null : spdxFile.CopyrightText,
+            LicenseConcluded = spdxFile.GetSingleLicense(RelationshipType.HAS_CONCLUDED_LICENSE, spdx30Elements, relationships),
+            LicenseInfoInFiles = spdxFile.GetMultipleLicenses(RelationshipType.HAS_DECLARED_LICENSE, spdx30Elements, relationships),
+            Path = spdxFile.Name,
+            Id = spdxFile.SpdxId
+        };
+
+        return sbomFile;
+    }
 
-        // Not setting LicenseConcluded and LicenseInfoInFiles since the whole SBOM is required to set these values.
-        return new SbomFile
+    public static SbomPackage ToSbomPackage(this Package spdxPackage, List<Element> spdx30Elements, List<Common.Spdx30Entities.Relationship> relationships)
+    {
+        var sbomPackage = new SbomPackage
         {
-            Checksum = checksums,
-            FileCopyrightText = spdxFile.CopyrightText,
-            Id = spdxFile.SpdxId,
-            Path = spdxFile.Name
+            PackageName = spdxPackage.Name,
+            PackageVersion = spdxPackage.PackageVersion,
+            PackageSource = spdxPackage.DownloadLocation == "NOASSERTION" ? null : spdxPackage.DownloadLocation,
+            CopyrightText = spdxPackage.CopyrightText == "NOASSERTION" ? null : spdxPackage.CopyrightText,
+            Checksum = spdxPackage.VerifiedUsing.ToSbomChecksum(),
+            LicenseInfo = new LicenseInfo
+            {
+                Concluded = spdxPackage.GetSingleLicense(RelationshipType.HAS_CONCLUDED_LICENSE, spdx30Elements, relationships),
+                Declared = spdxPackage.GetSingleLicense(RelationshipType.HAS_DECLARED_LICENSE, spdx30Elements, relationships),
+            },
+            Supplier = spdxPackage.GetSupplier(spdx30Elements),
+            PackageUrl = spdxPackage.GetPackageUrl(spdx30Elements),
+            FilesAnalyzed = true,
+            Id = spdxPackage.SpdxId,
+        };
+
+        return sbomPackage;
+    }
+
+    public static List<SbomRelationship> ToSbomRelationship(this Common.Spdx30Entities.Relationship spdxRelationship)
```

Comment:
What is the intention with all of these new methods? Are they eventually going to be called somewhere?

Reply:
Yup they will be called here - https://github.com/microsoft/sbom-tool/pull/1011

Reply:
I just wanted to separate the changes since it is a larger change